test/suite/test_checkpoint_snapshot05.py

```diff
@@ -0,0 +1,174 @@
+        # Open up the backup cursor, and copy the files.  Do a full backup.
+        cursor = self.session.open_cursor('backup:', None, None)
+        self.pr('Full backup from '+ fromdir + ' to ' + todir + ': ')
+        os.mkdir(todir)
+        while True:
+            ret = cursor.next()
+            if ret != 0:
+                break
+            bkup_file = cursor.get_key()
+            copy_file = os.path.join(fromdir, bkup_file)
+            sz = os.path.getsize(copy_file)
+            self.pr('Copy from: ' + bkup_file + ' (' + str(sz) + ') to ' + todir)
+            shutil.copy(copy_file, todir)
+        self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
+        cursor.close()
+
+    def check(self, check_value, uri, nrows):
+        session = self.session
+        session.begin_transaction()
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            if self.value_format == '8t':
+                self.assertEqual(v, check_value)
+            else:
+                self.assertEqual(v, check_value + str(count + 1))
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def evict(self, uri, ds, nrows):
+        s = self.conn.open_session()
+        s.begin_transaction()
+        # Configure debug behavior on a cursor to evict the page positioned on when the reset API is used.
+        evict_cursor = s.open_cursor(uri, None, "debug=(release_evict)")
+        for i in range(1, nrows + 1):
+            evict_cursor.set_key(ds.key(i))
+            self.assertEquals(evict_cursor.search(), 0)
+            evict_cursor.reset()
+        s.rollback_transaction()
+        evict_cursor.close()
+        s.close()
+
+    def test_checkpoint_snapshot(self):
+        self.moresetup()
+
+        ds = SimpleDataSet(self, self.uri, 0, \
+                key_format=self.key_format, value_format=self.value_format, \
+                config='log=(enabled=false)'+self.extraconfig)
+        ds.populate()
+
+        cursor = self.session.open_cursor(self.uri, None, "bulk")
+        for i in range(1, self.nrows + 1):
+            if self.value_format == '8t':
+                cursor[i] = self.valuea
+            else:
+                cursor[i] = self.valuea + str(i)
+        cursor.close()
+
+        self.check(self.valuea, self.uri, self.nrows)
+
+        session1 = self.conn.open_session()
+        session1.begin_transaction()
+        cursor1 = session1.open_cursor(self.uri)
+
+        for i in range(1, self.nrows + 1):
+            cursor1.set_key(ds.key(i))
+            if self.value_format == '8t':
+                cursor1.set_value(self.valueb)
+            else:
+                cursor1.set_value(self.valueb + str(i))
+            cursor1.set_value(self.valueb)
+            self.assertEqual(cursor1.update(), 0)
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
```

Comment:
This does not have the changes from WT-11492. Is that intentional?

Reply:
I must have missed them, sorry. I will double check everything, thank you for catching this!

Reply:
You should see modifications to `wtthread.py` and this test. I'll keep an eye out when you push (or you can ping me if you need quick turnaround) and can verify for you almost instantly.